src/encode/pattern/mod.rs

```diff
@@ -650,6 +688,42 @@ impl PatternEncoder {
         PatternEncoder {
             chunks: Parser::new(pattern).map(From::from).collect(),
             pattern: pattern.to_owned(),
+            color_map: default_color_map(),
+        }
+    }
+
+    /// Creates a `PatternEncoder` from a pattern string and color hashmap.
+    ///
+    /// The pattern string syntax is documented in the `pattern` module.
+    ///
+    ///
+    ///! ```no_run
+    ///! # #[cfg(all(feature = "console_appender",
+    ///! #           feature = "file_appender",
+    ///! #           feature = "pattern_encoder"))]
+    ///! # fn f() {
+    ///! // Change color of Info level msgs, other msgs will be the default color
+    ///! let mut log_color_map = HashMap::new();
+    ///! log_color_map.insert(log::Level::Info, Some(log4rs::encode::Color::Cyan));
+    ///! let pattern_encoder = PatternEncoder::new_with_colormap("{d} - {m}{n}", log_color_map);
+    ///!
+    ///! }
+    ///! # }
+    ///! # fn main() {}
+    ///! ```
+    pub fn new_with_colormap(
+        pattern: &str,
+        color_map: HashMap<Level, Option<Color>>,
+    ) -> PatternEncoder {
+        // Merge default color_map with user-configured color_map
+        let mut color_map_def = default_color_map();
+        for (k, v) in color_map {
+            color_map_def.insert(k, v);
+        }
+        PatternEncoder {
+            chunks: Parser::new(pattern).map(From::from).collect(),
```

Comment:
I think the preference is for Into::into

Reply:
You mean `Into::into` instead of `Parser::new`?

Reply:
I am mistaken about the guidance:
https://ricardomartins.cc/2016/08/03/convenient_and_idiomatic_conversions_in_rust

While From::from() focuses on the target type, Into::into() focuses on the original value; yet both express the same conversion. All the conversions above are equivalent, choosing one of them is a matter of taste. Personally, I prefer using TargetType::from(value) and value.into(). The former makes our intention clearer, while the latter is shorter than Into::into(value). Note that we might need to add type annotations to disambiguate the intended target type if we opt any form other than TargetType::from(), which clearly indicates it.